File: API/Helpers/UserParams.cs
namespace API.Helpers
{
    public class UserParams
    {
        private const int MaxPageSize = 40;
        private int pageNumber = 1;
        public int PageNumber { 
            get {
                return pageNumber; 
            }
            set {
                if(value > 0)
                {
                    pageNumber = value;
                }
            } 
        }
        private int itemsPerPage;
        public int ItemsPerPage { 
            get {
                return itemsPerPage;
            }
            set { 
                if(value > MaxPageSize || value < 1)
                {
                    itemsPerPage = MaxPageSize;
                }
                else
                {
                    itemsPerPage = value;
                }
            } 
        }
    }
}

File: WebUI/src/app/_models/userParams.ts
import { User } from "./User";

export class UserParams{
    minAge :  number = 18;
    maxAge : null | number = null;
    pageNumber : number = 1;
    itemsPerPage: number = 4;
    orderBy: string = 'lastActive'; // options lastActive, age, creationTime
    sex: string;
    constructor( user: User){
        this.sex = user.interest;
    }
}